src/menu/node-menu.component.ts

```diff
@@ -62,12 +64,13 @@ export class NodeMenuComponent implements OnInit, OnDestroy {
   public onMenuItemSelected(e: MouseEvent, selectedMenuItem: NodeMenuItem): void {
     if (isLeftButtonClicked(e)) {
       this.menuItemSelected.emit({nodeMenuItemAction: selectedMenuItem.action});
+      this.nodeMenuService.fireMenuEvent(e.target as HTMLElement, NodeMenuAction.Close);
     }
   }
 
   private closeMenu(e: MouseEvent | KeyboardEvent): void {
     const mouseClicked = e instanceof MouseEvent;
-    if (mouseClicked || isEscapePressed(e as KeyboardEvent)) {
+    if (mouseClicked && !(this.menuContainer.nativeElement !== e.target && this.menuContainer.nativeElement.contains(e.target)) || isEscapePressed(e as KeyboardEvent)) {
```

Comment:
This guy is huge :) Let's extract this expression `!(this.menuContainer.nativeElement !== e.target && this.menuContainer.nativeElement.contains(e.target))` and give it a name.